packages/sidenav/src/ux-sidenav/ux-sidenav.ts

```diff
@@ -0,0 +1,107 @@
+import { customElement, useView, inject, TaskQueue, PLATFORM, bindable } from "aurelia-framework";
+import { UxSidenavDrawer, OPEN_CHANGED } from "../ux-sidenav-drawer/ux-sidenav-drawer";
+import { UxSidenavTheme } from "./ux-sidenav-theme";
+import { StyleEngine, normalizeNumberAttribute } from "@aurelia-ux/core";
+import { IUxSidenavDrawerElement } from "../ux-sidenav-drawer/i-ux-sidenav-drawer-element";
+import { UxDefaultSidenavConfiguration } from "../ux-default-sidenav-configuration";
+
+@inject(Element, StyleEngine, TaskQueue, UxDefaultSidenavConfiguration)
+@customElement('ux-sidenav')
+@useView(PLATFORM.moduleName('./ux-sidenav.html'))
+export class UxSidenav {
+  constructor(public element: HTMLElement, private styleEngine: StyleEngine, private taskQueue: TaskQueue,
+    private defaultConfig: UxDefaultSidenavConfiguration) { }
+
+  backdrop: boolean = this.defaultConfig.backdrop;
+  width: number;
+
+  leftDrawer: UxSidenavDrawer | undefined;
+  rightDrawer: UxSidenavDrawer | undefined;
+  bottomDrawer: UxSidenavDrawer | undefined;
+
+  private _modalBreakpoint: number = this.defaultConfig.modalBreakpoint;
+  @bindable
+  public modalBreakpoint: number | string = this.defaultConfig.modalBreakpoint;
+  modalBreakpointChanged() {
+    this._modalBreakpoint = normalizeNumberAttribute(this.modalBreakpoint) ?? 0;
+  }
+
+  @bindable
+  public theme: UxSidenavTheme;
+  public themeChanged(newValue: UxSidenavTheme) {
+    if (newValue !== null && !newValue.themeKey) {
+      newValue.themeKey = 'sidenav';
+    }
+
+    this.styleEngine.applyTheme(newValue, this.element);
+  }
+
+  attached() {
+    this.leftDrawer = this.element.querySelector<IUxSidenavDrawerElement>('ux-sidenav-drawer[side="left"]')?.au.controller.viewModel;
+    this.rightDrawer = this.element.querySelector<IUxSidenavDrawerElement>('ux-sidenav-drawer[side="right"]')?.au.controller.viewModel;
+    this.bottomDrawer = this.element.querySelector<IUxSidenavDrawerElement>('ux-sidenav-drawer[side="bottom"]')?.au.controller.viewModel;
```

Comment:
Could we make this more flexible ? I guess that discovering the drawers on `attached` will be fine for 95% of usages but do not cover edge cases where one would want to have drawer elements behind `if.bind` ?

Wouldn't it be better to use `@processContent` of `@children` decorators to discover the drawers ?

@bigopon any thoughts ?

Reply:
It works under if.bind, check out theming page. What does NOT work is @child under if.bind. It never gets populated

Reply:
I'll be fixing this one over this weekend. For now, this is probably the way we have to do it 😓

Reply:
You mean that it will work with this ? 

```html
<ux-sidenav class="theming">
  <ux-sidenav-drawer side="left" if.bind="shouldDisplayDrawer">Drawer</ux-sidenav-drawer>
  <!-- ... -->
</ux-sidenav>
```

Didn't see this in the theming page.

Reply:
here is the PR that fixes this issue https://github.com/aurelia/templating/pull/685
@ben-girardet yes, it's a subtle bug, only when you put the if on the parent and hide/show via an if.